[aotinductor] Add a perf smoke test for AOTInductor

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110972

Summary: To prevent perf regression like the one caused by #110510

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

desertfire committed:
[aotinductor] Add a perf smoke test for AOTInductor

Summary: To prevent perf regression like the one caused by #110510

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add a perf smoke test for AOTInductor"


Summary: To prevent perf regression like the one caused by #110510

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add a perf smoke test for AOTInductor"


Summary: To prevent perf regression like the one caused by #110510

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

chenyang78 commented on .ci/pytorch/test.sh:
```diff
@@ -485,9 +485,13 @@ test_inductor_torchbench_smoketest_perf() {
+  python benchmarks/dynamo/torchbench.py --device cuda --performance --bfloat16 --inference \
+    --export-aot-inductor --only nanogpt --output "$TEST_REPORTS_DIR/inductor_inference_smoketest.csv"
+  # The threshold value needs to be actively maintained to make this check useful
+  python benchmarks/dynamo/check_perf_csv.py -f "$TEST_REPORTS_DIR/inductor_inference_smoketest.csv" -t 5.25
```
Out of curiosity, how is the value ```5.25``` being determined, by some estimation? Thanks.

desertfire replied:
I took 95% of the value reported on the OSS dashboard.

desertfire committed:
Update on "[aotinductor] Add a perf smoke test for AOTInductor"


Summary: To prevent perf regression like the one caused by #110510

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "add perf smoke test has passed"
